src/components/TextField/TextField.tsx

```diff
@@ -405,6 +418,7 @@ export function TextField({
     maxLength,
     spellCheck,
     pattern,
```

Comment:
I’d argue this should default to `[0-9]*` on `number` inputs to show the right keyboard. It has better support than `inputMode`.